src/compiler/parser.ts

```diff
@@ -5938,7 +5939,18 @@ namespace Parser {
                 nextToken(); // advance past the 'import'
                 nextToken(); // advance past the dot
                 expression = finishNode(factory.createMetaProperty(SyntaxKind.ImportKeyword, parseIdentifierName()), pos);
-                sourceFlags |= NodeFlags.PossiblyContainsImportMeta;
+
+                if ((expression as MetaProperty).name.escapedText === "defer") {
+                    if (token() === SyntaxKind.OpenParenToken || token() === SyntaxKind.LessThanToken) {
+                        sourceFlags |= NodeFlags.PossiblyContainsDynamicImport;
+                    }
+                    else {
+                        parseErrorAt(pos, getNodePos(), Diagnostics.import_defer_must_be_followed_by);

```

Comment:
Normally we would just issue a `'(' expected` diagnostic. We also may want to treat this as a grammar error (in the checker) rather than a parse error so as not to turn off other semantic checks in the file.